aiounifi/devices.py

```diff
@@ -345,3 +385,76 @@ def __getitem__(self, obj_id: int) -> Port:
     def __iter__(self) -> Iterator[Union[int, str]]:
         """Iterate over ports."""
         return iter(self.ports)
+
+
+class Outlet:
+    """Represents an outlet."""
+
+    def __init__(self, raw: dict) -> None:
+        """Initialize outlet."""
+        self.raw = raw
+
+    @property
+    def name(self) -> str:
+        """Name of outlet"""
+        return self.raw["name"]
+
+    @property
+    def index(self) -> int:
+        """Outlet index."""
+        return self.raw["index"]
+
+    @property
+    def has_relay(self) -> bool:
+        """Is the outlet controllable"""
+        return self.raw["has_relay"]
+
+    @property
+    def has_metering(self) -> bool:
+        """Is metering supported."""
+        return self.raw["has_metering"]
+
+    @property
+    def relay_state(self) -> Optional[bool]:
+        """Is outlet power on."""
+        return self.raw.get("relay_state")
+
+    @property
+    def cycle_enabled(self) -> Optional[bool]:
+        """Modem Power Cycle"""
+        return self.raw.get("cycle_enabled")
+
+
+class Outlets:
+    """Represents outlets on a device."""
+
+    def __init__(self, raw: List[dict]) -> None:
+        """Initialize outlet manager."""
+        self.outlets: Dict[int, Outlet] = {}
+        for raw_outlet in raw:
+            outlet = Outlet(raw_outlet)
+            self.outlets[outlet.index] = outlet
+
+
+    def update(self, raw: List[dict]) -> None:
+        """Update outlets."""
+        for raw_outlet in raw:
+            index = None
+
+            if "index" in raw_outlet:
+                index = raw_outlet["index"]
+
+            if index in self.outlets:
+                self.outlets[index].raw = raw_outlet
```

Comment:
We're saying we expect index to exist right so lets just do this
```suggestion
            index = raw_outlet["index"]
            self.outlets[index].raw = raw_outlet
```